.github/workflows/ci.yml

```diff
@@ -0,0 +1,35 @@
+---
+
+name: CI
+
+on:
+  pull_request:
+    paths:
+      - "src/**"
+      - "tests/**"
+  workflow_dispatch: {}
+
+jobs:
+  test:
+    name: CI workflow-linter (v2)
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - name: Set up Python
+        uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
+        with:
+          python-version: "3.11"
```

Comment:
🎨 To be more DRY it would be nice to see a `.python-version` file controlling all this.